change Button variant types constant

aswathkk opened the pull request:
## Description

Changes the constant `ButtonVariantTypes` from SOLID, OUTLINE, GHOST to PRIMARY, SECONDARY and TERTIARY

Fixes #7793 

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

Manually tested

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

somangshu reviewed (approved):
Approving for tests to run

somangshu reviewed (commented):
@aswathkk did this not have any related tests, Do we need to write any? 

The change looks good to me, Still need to check the deploy preview. Meanwhile can you merge the latest release here? There is an unrelated test failing continuously

somangshu commented on app/client/src/components/formControls/FilePickerControl.tsx:
```diff
@@ -102,7 +103,7 @@ class FieldFileInput extends React.Component<Props, FieldFileInputState> {
         <SelectButton
           buttonStyle="PRIMARY"
-          buttonVariant="OUTLINE"
+          buttonVariant={ButtonVariantTypes.SECONDARY}
```
QQ: the variant given here is secondary, but the file-picker button still is showing up as primary. What are we trying to do here?

aswathkk replied:
@somangshu This code is related to the data source config page. Not the file-picker widget

![Screenshot 2021-09-30 at 4 28 18 PM](https://user-images.githubusercontent.com/10436935/135443154-a667fcbd-fa57-4cb5-bf8a-c8d404f72fc0.png)

I just changed the hard-coded string to a constant.
Should I revert this change since it is not related to UI Building?

somangshu commented:
@aswathkk can you test if we need migration here, I believe that old apps will break if they have used the JS expression with any older values?

aswathkk commented:
@somangshu We need migrations here. I'm working on it.
Do you think we need to write any other tests other than tests for the migrations?

somangshu commented on app/client/src/widgets/ButtonWidget/component/index.tsx:
```diff
@@ -265,6 +265,7 @@ export function BaseButton(props: IButtonProps & ButtonStyleProps) {
         buttonColor={buttonColor}
         buttonVariant={buttonVariant}
         className={className}
+        data-test-variant={buttonVariant}
```
Why is this required @aswathkk? Cant we just use the buttonVariant property?

aswathkk replied:
@somangshu We are omitting `buttonVariant` property on line number 143. So, that property is not available in the DOM.

somangshu reviewed (approved):
Approving to run the tests

somangshu commented:
Some tests are failing, Can you check the dashboard @aswathkk

aswathkk commented:
@somangshu Tests passed initially. But when I responded to your comment, it Skipped.

somangshu reviewed (approved):
Approving again to run the workflows

aswathkk commented:
@somangshu Can we do QA for this?

somangshu commented:
@aswathkk please have details about the change here for the QA, point out the migration that needs to be tested. We will need to be cautious here

aswathkk commented:
@vivekverma2312 We changed the ButtonVariant constant for the following widgets:
- ButtonWidget
- MenuWidget
- FormButtonWidget
- IconButtonWidget
And also for the button that you see on TableWidget. Migrations are also added for these.

Changes are as follows:
SOLID --> PRIMARY
OUTLINE --> SECONDARY
GHOST --> TERTIARY

To see this, you need to change the variant to JSEditatble.

YogeshJayaseelan commented:
@aswathkk - Issue 1 - For the Table Icon and Menu Button - When the Button variant is bind-ed with text widget with SOLD, GHOST or OUTLINE - No error message is displayed 

https://www.loom.com/share/612e3474b7e3479e91c26d1f89ebc8dc

somangshu commented:
@aswathkk @YogeshJayaseelan have we tested if any old apps fail with the changes in the table widget? Do we have the button variants in PROD in any of the column types? Lets make sure!
